### ISSUE TITLE
Why take the opposite direction when generating a point cloud in kitti_utils.py?

### ISSUE THREAD

**Issue by MaxChanger** [laoreja/HPLFlowNet#22]
Hello @laoreja, I used the scripts you provided to process the KITTI dataset and it could work successfully, but I have a little confused about **#Line 25**.  I found the problem because I was visualizing the point cloud, and there were some errrors in the rotation, so why do you take the negative of x and y here (the opposite direction)

https://github.com/laoreja/HPLFlowNet/blob/9a4fdca36f2479cc560bc0c362c7db6183d66c28/data_preprocess/kitti_utils.py#L21-L26

Could you give me some hints about this, thank you very much.

> The default Camera system should be the x axis right, the y axis down, and the z axis forward. Am I right?

**Comment by laoreja** [comment]
Hi,
I think this processing is to make sure that the KITTI point clouds are preprocessed similarly as FlyingThings3D.
Since we only train on FlyingThings3D, and directly test on KITTI.

In general, as long as your training data and your test data are distributed in the same orientation, then that's fine.

Thank you!

**Comment by MaxChanger** [comment]
Thanks @laoreja, but how does FlyingThings3D define the orientation? In **#L21-33** you have the code for converting that disparity to PointCloud as follows
https://github.com/laoreja/HPLFlowNet/blob/9a4fdca36f2479cc560bc0c362c7db6183d66c28/data_preprocess/flyingthings3d_utils.py#L21-L33
In **#L27**, why is it multiplied by -1? And y doesn't have any operations.
Another, in the official website of FlyingThings3D, ` Note that disparities for both views are stored as positive numbers.` but when I read *.pfm through this code, they are all negative. Why is that? I'm a little confused.

In your settings, f is negative, disparity is negative and then the depth is negative. This may be a bit abnormal, the point cloud should be in front of the camera under normal circumstances.

Thanks,  hope you can explain this part

**Comment by MaxChanger** [comment]
@laoreja Any replies?

**Comment by MaxChanger** [comment]
I have discussed with the kindly author via email. 
Although the axis of FlyingThings is a bit elusive in the data generation stage, the same distribution as KITTI is guaranteed in the dataloader stage, as following:
https://github.com/laoreja/HPLFlowNet/blob/9a4fdca36f2479cc560bc0c362c7db6183d66c28/datasets/flyingthings3d_subset.py#L93-L99

Thanks again.
